### PR TITLE
David Barriga-Gomez

### PR DESCRIPTION
**Pull request by DavidBarriga-Gomez**

### CODE REVIEW

File: test/room_test.rb [hunk 1]
```diff
@@ -4,4 +4,19 @@
...
+    assert_instance_of Room, @room
+  end
+
+  def test_if_room_has_a_category
+    assert_equal :bedroom, @room.category
+  end
+
+  def test_if_room_has_area
+    assert_equal 130, @room.area
+  end
```
Comment (iandouglas) [inline test/room_test.rb]: great job testing in here

File: test/house_test.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+class HouseTest < Minitest::Test
+
+  def setup
+    @house = House.new("$400000", "123 sugar lane")
+    @room_1 = Room.new(:bedroom, 10, 13)
+    @room_2 = Room.new(:bedroom, 11, 15)
+    @room_3 = Room.new(:living_room, 25, 15)
+    @room_4 = Room.new(:basement, 30, 41)
+  end
```
Comment (iandouglas) [inline test/house_test.rb]: great job with your setup in here

File: test/house_test.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+  def test_if_it_exists
+    assert_instance_of House, @house
+  end
+
+  def test_if_it_has_price
+    assert_equal "$400000", @house.price
```
Comment (iandouglas) [inline test/house_test.rb]: be careful to pay very close attention to our interaction patterns, we wanted the price to be returned as an integer, not the original string:
```ruby
house = House.new("$400000", "123 sugar lane")
pry(main)> house.price
#=> 400000
```

File: test/house_test.rb [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  end
+
+  def test_if_it_has_address
+    assert_equal "123 sugar lane", @house.address
+  end
+
```
Comment (iandouglas) [inline test/house_test.rb]: good testing in here so far, it would also be good to test that the house starts out with an empty `rooms` array too

File: test/house_test.rb [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    assert_equal [@room_1, @room_2], @house.rooms
+  end
+
+  def test_rooms_from_category
+    @house.add_rooms(@room_1)
+    @house.add_rooms(@room_2)
+    @house.add_rooms(@room_3)
+    @house.add_rooms(@room_4)
+    assert_equal [@room_1, @room_2], @house.rooms_from_category(:bedroom)
+    assert_equal [@room_4], @house.rooms_from_category(:basement)
```
Comment (iandouglas) [inline test/house_test.rb]: good job in here, remember for readability to add a blank line between any additional setup needed in the test and your assertions

File: test/house_test.rb [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  end
+
+  def test_area_of_entire_house
+    @house.area(@room_1)
+    @house.area(@room_2)
+    @house.area(@room_3)
+    @house.area(@room_4)
+    assert_equal 1900, @house.area_of_house
```
Comment (iandouglas) [inline test/house_test.rb]: nice work here

File: lib/house.rb [hunk 7]
```diff
@@ -0,0 +1,28 @@
+class House
+  attr_reader :price, :address, :rooms, :area
```
Comment (iandouglas) [inline lib/house.rb]: remember that we only need attr_readers for instance variables that we intend to expose; in this case `area` would be our method below, not an instance variable

File: lib/house.rb [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+  def initialize(price, address)
+    @price = price
+    @address = address
+    @rooms = []
+    @spaces = []
```
Comment (iandouglas) [inline lib/house.rb]: be careful adding extra instance variables that might be one-time-use; this might be better defined in the method below where you use it

File: test/house_test.rb [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+  def test_if_it_has_price
+    assert_equal "$400000", @house.price
+  end
+
+  def test_if_it_has_address
+    assert_equal "123 sugar lane", @house.address
+  end
+
+  def test_if_house_add_rooms
+    @house.add_rooms(@room_1)
```
Comment (iandouglas) [inline test/house_test.rb]: be careful of your method naming here, as we wanted this to be `add_room`, and `add_rooms` (plural) implies that we could pass multiple rooms into the method

File: lib/house.rb [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+
+  end
+
+  def add_rooms (new_rooms)
```
Comment (iandouglas) [inline lib/house.rb]: per my earlier note, be careful to name methods properly; in this case, pluralizing the name of the method implies that we might be adding multiple rooms at a time to our house

File: lib/house.rb [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+    @rooms.push (new_rooms)
+  end
+
+  def rooms_from_category(category)
+    @rooms.find_all do |room|
+    room.category == category
+  end
+  end
```
Comment (iandouglas) [inline lib/house.rb]: be careful on your indentation here so that the code is more readable; good job on the logic here, you can also condense this to one line: `@rooms.find_all {|room| room.category == category}`

File: lib/house.rb [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+
+  def area_of_house
+    @spaces.map do |space|
+      # +=
+  end
+  #   @room_1.area + @room_2.area + @room_3.area + @room_4.area
+  # enumerable
```
Comment (iandouglas) [inline lib/house.rb]: good start here; you would have wanted to iterate over all of your `@rooms` again and add up their `area` calls. There's a way to use the `.sum` method here with a block of code, like `@rooms.sum {|room| room.area}` to calculate this

File: lib/room.rb [hunk 13]
```diff
@@ -1,3 +1,12 @@
 class Room
+  attr_reader :category
+  def initialize(category, width, length)
+    @category, = category
+    @width = width
+    @length = length
+  end
 
+  def area
+    @width * @length
```
Comment (iandouglas) [inline lib/room.rb]: good job doing this here; this kind of calculation is sort of a one-time calculation, so you could move this into your initialize method like `@area = width * length` and then expose `@area` as another attr_reader
